h2/src/main/org/h2/command/dml/Select.java

```diff
@@ -1460,6 +1460,7 @@ public void reset() {
                     }
                     return row;
                 }
+                --rowNumber;
```

Comment:
perhaps a better fix would be to move the 
    setCurrentRowNumber(++rowNumber);
inside the 
  if (isConditionMet())
block
   ????

Reply:
I assume that isConditionMet() can potentially access currentRowNumber, so increment inside block will not work - too late, but it can be done the same way as in LazyResultGroupSorted.fetchNextRow(), though. Just to be consistent, I'll fix it later today, have no access right now.